### PR TITLE
Add a doc section about stuck master scale down in ES <= 7.8

### LINKED ISSUE #1986: Do not try to exclude a master node that never existed
I ran out of resources on a K8S cluster while doing an upscale of a set of MDI nodes.

I'm now in a situation where the `nodeSet` can't be downscaled because the operator is trying to exclude a master node  which has never existed:

```
pod/es-apm-sample-es-1-0                    1/1     Running   0          33m    10.56.3.18    gke-michael-dev-cluster-default-pool-8a982915-3msd   <none>           <none>
pod/es-apm-sample-es-1-1                    0/1     Pending   0          26m    <none>        <none>
```

```
2019-10-14T07:30:50.277Z	ERROR	controller-runtime.controller	Reconciler error
{
	"ver": "1.0.0-beta1-bc11-c8bb5e5b",
	"controller": "elasticsearch-controller",
	"request": "default/es-apm-sample",
	"error": "unable to add to voting_config_exclusions: 400 Bad Request: add voting config exclusions request for [es-apm-sample-es-1-1] matched no master-eligible nodes",
	"errorCauses": [{
		"error": "unable to add to voting_config_exclusions: 400 Bad Request: unknown",
		"errorVerbose": "400 Bad Request: unknown
unable to add to voting_config_exclusions
github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client.(*clientV7).AddVotingConfigExclusions\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client/v7.go:41\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/version/zen2.AddToVotingConfigExclusions\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/version/zen2/voting_exclusions.go:34\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver.updateZenSettingsForDownscale\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver/downscale.go:237\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver.doDownscale\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver/downscale.go:198\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver.attemptDownscale\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver/downscale.go:129\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver.HandleDownscale\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver/downscale.go:54\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver.(*defaultDriver).reconcileNodeSpecs\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver/nodes.go:112\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver.(*defaultDriver).Reconcile\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/driver/driver.go:234\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch.(*ReconcileElasticsearch).internalReconcile\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/elasticsearch_controller.go:284\ngithub.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch.(*ReconcileElasticsearch).Reconcile\n\t/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/elasticsearch_controller.go:219\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.1/pkg/internal/controller/controller.go:216\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.1/pkg/internal/controller/controller.go:192\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.1/pkg/internal/controller/controller.go:171\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1357"
	}]
}
```

### PR DESCRIPTION
**Pull request by pebrc**
Fixes #1986 

Preview: https://cloud-on-k8s_4078.docs-preview.app.elstc.co/guide/en/cloud-on-k8s/master/k8s-common-problems.html#k8s-common-problems-scale-down

### CODE REVIEW

**Review by barkbay — APPROVED** [review]
LGTM 👍

File: docs/operating-eck/troubleshooting/common-problems.asciidoc [hunk 1]
```diff
@@ -85,6 +85,38 @@ type: Opaque
 
 Failure to do so can cause data loss.
 
+[id="{p}-{page_id}-scale-down"]
+== Scale down of Elasticsearch master-eligible Pods seems stuck
+
+If a master-eligible Elasticsarch Pod was never successfully scheduled and the Elasticsearch cluster is running version 7.8 or earlier ECK may fail to scale down the Pod. To find out whether you are affected, check that the Pod in question is pending:
```
Comment (alaudazzi) [inline docs/operating-eck/troubleshooting/common-problems.asciidoc:91]:
```suggestion
If a master-eligible Elasticsarch Pod was never successfully scheduled and the Elasticsearch cluster is running version 7.8 or earlier, ECK may fail to scale down the Pod. To find out whether you are affected, check if the Pod in question is pending:
```

File: docs/operating-eck/troubleshooting/common-problems.asciidoc [hunk 2]
```diff
@@ -85,6 +85,38 @@ type: Opaque
...
+pod/<cluster-name>-es-<nodeset>-1                    0/1     Pending   0          26m    <none>        <none>
+----
+
+Check the <<{p}-get-eck-logs,operator logs>> for an error similar to:
+[source,sh]
+----
+"unable to add to voting_config_exclusions: 400 Bad Request: add voting config exclusions request for [<cluster-name>-es-<nodeset>-1] matched no master-eligible nodes",
+----
+
+To work around this issue scale down the underlying StatefulSet manually. First identify the affected StatefulSet and the number of Pods that are ready (symbolized by `m` in this example):
```
Comment (alaudazzi) [inline docs/operating-eck/troubleshooting/common-problems.asciidoc:104]:
```suggestion
To work around this issue, scale down the underlying StatefulSet manually. First, identify the affected StatefulSet and the number of Pods that are ready (symbolized by `m` in this example):
```

File: docs/operating-eck/troubleshooting/common-problems.asciidoc [hunk 3]
```diff
@@ -85,6 +85,38 @@ type: Opaque
...
+
+[source,sh]
+----
+> kubectl get sts -l elasticsearch.k8s.elastic.co/cluster-name=<cluster-name>
+NAME                       READY   AGE
+<cluster-name>-es-<nodeset>   m/n     44h
+----
+Then scale down the StatefulSet to the right size `m`, removing the pending Pod:
```
Comment (alaudazzi) [inline docs/operating-eck/troubleshooting/common-problems.asciidoc:112]:
```suggestion
Then, scale down the StatefulSet to the right size `m`, removing the pending Pod:
```

File: docs/operating-eck/troubleshooting/common-problems.asciidoc [hunk 4]
```diff
@@ -85,6 +85,38 @@ type: Opaque
...
+[source,sh]
+----
+> kubectl scale --replicas=m  sts/<cluster-name>-es-<nodeset>
+----
+
+CAUTION: Do not use this method to scale down Pods that have already joined the Elasticsearch cluster as additional data loss protection that ECK applies is sidestepped.
```
Comment (alaudazzi) [inline docs/operating-eck/troubleshooting/common-problems.asciidoc:118]:
```suggestion
CAUTION: Do not use this method to scale down Pods that have already joined the Elasticsearch cluster, as additional data loss protection that ECK applies is sidestepped.
```

**Review by alaudazzi — APPROVED** [review]
Left a few minor suggestions, mainly punctuation. 
LGTM.

**Pull request merged**